webapp/components/survey/CategorySelector/CategorySelector.js

```diff
@@ -28,7 +29,7 @@ const CategorySelector = (props) => {
   const [showCategoriesPanel, setShowCategoriesPanel] = useState(false)
 
   const survey = useSurvey()
-  const categories = Survey.getCategoriesArray(survey)
+  const surveyId = useSurveyId()
   const category = Survey.getCategoryByUuid(categoryUuid)(survey)
```

Comment:
category won't belong to survey anymore. it must be fetched on mount

Reply:
check that for taxonomies as well

Reply:
Into the next commit I added a useSurveyTaxonomy Hook, can u take a look on that and check it is can be a good approach?, also if it is, where is the right place to place* that?

Reply:
also, we can move the lookup functions inside of this hooks